nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkComputation.java

```diff
@@ -70,14 +71,20 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
                 new WorkHolder(work).run();
                 workIds.add(work.getId());
             }
+            work.cleanUp(true, null);
+            context.askForCheckpoint();
         } catch (Exception e) {
-            // TODO: check what to catch exactly we don't want to kill the computation on error
-            log.warn(String.format("Work id: %s title: %s, raise an exception, work is marked as completed",
-                    work.getId(), work.getTitle()), e);
+            if (ExceptionUtils.hasInterruptedCause(e)) {
+                Thread.currentThread().interrupt();
```

Comment:
Maybe also add a comment explaining that because the interrupted state is set again, the loop processing in `processLoop()` will be stopped because `continueLoop()` will return `false`.